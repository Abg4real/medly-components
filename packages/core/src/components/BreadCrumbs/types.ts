import { WithThemeProp } from '@medly-components/utils';

export interface StyledProps extends WithThemeProp {
    bgColor: string;
    textColor: string;
}
