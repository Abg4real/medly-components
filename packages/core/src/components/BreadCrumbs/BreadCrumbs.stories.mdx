import { BreadCrumbs } from './BreadCrumbs.tsx';
import { defaultTheme } from '@medly-components/theme';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import * as stories from './BreadCrumbs.stories';

<Meta title="Core/BreadCrumbs" component={BreadCrumbs} />

# Avatar

### Displays an image or text that represents a user or organization

<Preview withToolbar>
    <Story name="Basic">{stories.Basic}</Story>
</Preview>

### Props

<Props of={BreadCrumbs} />

### BreadCrumbsTheme

<Props of={stories.ThemeInterface} />

### BreadCrumbsTheme Defaults

<code>{JSON.stringify(defaultTheme.breadCrumbs, null, 2)}</code>
