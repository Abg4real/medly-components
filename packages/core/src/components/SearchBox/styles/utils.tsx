import { SvgIcon } from '@medly-components/icons';
import { css } from '@medly-components/utils';
import { Props } from '../types';

export const getSearchBoxSize = ({ theme, size }: Props) => theme.searchBox.sizes[size].height;

export const getBorderAndBoxShadow = (color: string, boxShadow: string) => {
    return css`
        border-color: ${color};
        box-shadow: ${boxShadow};
    `;
};

export const getIconWrapperStyle = ({ theme, size }: Props) => {
    return css`
        width: ${getSearchBoxSize};
        min-width: ${getSearchBoxSize};
        height: auto;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        ${SvgIcon} {
            cursor: pointer;
            font-size: ${theme.searchBox.sizes[size].iconSize};
        }
    `;
};
