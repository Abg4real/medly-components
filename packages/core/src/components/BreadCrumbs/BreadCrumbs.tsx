import { WithStyle } from '@medly-components/utils';
import React from 'react';
import List from '../List';
import { BreadCrumbsStyled } from './BreadCrumbs.styled';

const BreadCrumbs: React.SFC & WithStyle = React.memo(props => {
    return (
        <BreadCrumbsStyled {...props}>
            <List variant="horizontal">{props.children}</List>
        </BreadCrumbsStyled>
    );
});

BreadCrumbs.displayName = 'BreadCrumbs';
BreadCrumbs.Style = BreadCrumbsStyled;

export default BreadCrumbs;
