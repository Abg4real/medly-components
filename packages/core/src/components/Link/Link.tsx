import { isValidStringOrNumber, WithStyle } from '@medly-components/utils';
import React, { SFC } from 'react';
import Text from '../Text';
import { LinkStyled } from './Link.styled';
import { Props } from './types';

export const Link: SFC<Props> & WithStyle = React.memo(
    React.forwardRef(({ href, ...props }, ref) => {
        return (
            <LinkStyled href={href} {...props} ref={ref}>
                {React.Children.map(props.children, c => {
                    return isValidStringOrNumber(c) ? <Text>{c}</Text> : c;
                })}
            </LinkStyled>
        );
    })
);

Link.displayName = 'Link';
Link.Style = LinkStyled;
