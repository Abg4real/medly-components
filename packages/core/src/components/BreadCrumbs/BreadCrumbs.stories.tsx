import { BreadCrumbsTheme } from '@medly-components/theme';
import React from 'react';
import Link from '../Link';
import { BreadCrumbs } from './BreadCrumbs';

export const ThemeInterface = (props: BreadCrumbsTheme): any => null;

export const Basic = () => (
    <BreadCrumbs>
        <Link href="#">Covered Entities</Link>
        <Link href="#">Add New Entities</Link>
    </BreadCrumbs>
);
