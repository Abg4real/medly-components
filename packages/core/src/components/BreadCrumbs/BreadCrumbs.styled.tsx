import { defaultTheme } from '@medly-components/theme';
import { styled } from '@medly-components/utils';
import Link from '../Link';
import List from '../List';
import { StyledProps } from './types';

export const BreadCrumbsStyled = styled('div').attrs(({ theme: { breadCrumbs } }) => ({ ...breadCrumbs }))<StyledProps>`
    height: 40px;
    background-color: ${({ bgColor }) => bgColor};
    display: flex;
    align-items: center;

    ${Link.Style} {
        color: ${({ textColor }) => textColor};
    }

    ${List.Style} {
        li + li::before {
            margin-right: ${({ theme }) => theme.spacing.S2};
            content: '/';
            color: ${({ textColor }) => textColor};
        }
    }
`;

BreadCrumbsStyled.defaultProps = {
    theme: defaultTheme
};
