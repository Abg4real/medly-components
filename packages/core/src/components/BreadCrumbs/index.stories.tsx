import { storiesOf } from '@storybook/react';
import React from 'react';
import Link from '../Link';
import { BreadCrumbs } from './BreadCrumbs';

storiesOf('Core', module).add(
    'BreadCrumbs',
    () => (
        <BreadCrumbs>
            <Link href="#">Covered Entities</Link>
            <Link href="#">Add New Entities</Link>
        </BreadCrumbs>
    ),
    {
        props: {
            propTablesExclude: [Link]
        }
    }
);
