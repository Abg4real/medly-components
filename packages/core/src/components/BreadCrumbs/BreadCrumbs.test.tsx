import { TestUtils } from '@medly-components/utils';
import React from 'react';
import Link from '../Link';
import { BreadCrumbs } from './BreadCrumbs';

describe('BreadCrumbs component', () => {
    it('should render correctly', () => {
        const { container } = TestUtils.render(
            <BreadCrumbs>
                <Link href="#">Covered Entities</Link>
                <Link href="#">Add New Entities</Link>
            </BreadCrumbs>
        );
        expect(container).toMatchSnapshot();
    });
});
