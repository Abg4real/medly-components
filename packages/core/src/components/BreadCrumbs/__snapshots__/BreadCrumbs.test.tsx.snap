// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs component should render correctly 1`] = `
.c5 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c4 {
  color: #012040;
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:active {
  color: #012040;
}

.c4:hover {
  color: #012040;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  outline-width: 0;
  cursor: pointer;
}

.c2 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 li {
  margin: 0 0.4rem;
}

.c2 li:last-child {
  margin-right: 0.8rem;
}

.c2 li:first-child {
  margin-left: 0.8rem;
}

.c0 {
  height: 40px;
  background-color: #012040;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .c3 {
  color: #ffffff;
}

.c0 .c1 li + li::before {
  margin-right: 0.8rem;
  content: '/';
  color: #ffffff;
}

<div>
  <div
    class="c0"
  >
    <ul
      class="c1 c2"
    >
      <li>
        <a
          class="c3 c4"
          href="#"
        >
          <span
            class="c5"
          >
            Covered Entities
          </span>
        </a>
      </li>
      <li>
        <a
          class="c3 c4"
          href="#"
        >
          <span
            class="c5"
          >
            Add New Entities
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
